### PR TITLE
executor: support batch commit the DMLs

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Make all the DMLs could be batch committed.

### What is changed and how it works?
Add a config to control to split DMLs.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

### CODE REVIEW

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

**Comment by jackysp** [comment]
/run-all-tests tidb-test=release-2.0 tikv=release-2.0 pd=release-2.0

File: executor/write.go [hunk 1]
```diff
@@ -2090,3 +2091,17 @@ func (e *UpdateExec) Close() error {
 func (e *UpdateExec) Open(ctx context.Context) error {
 	return e.SelectExec.Open(ctx)
 }
+
+func batchDMLIfNeeded(ctx sessionctx.Context, count int) error {
+	if count%ctx.GetSessionVars().DMLBatchSize == 0 && count != 0 {
+		if err := ctx.StmtCommit(); err != nil {
+			return ErrBatchDMLFail.GenByArgs(err)
```
Comment (tiancaiamao) [inline executor/write.go]: `StmtRollback` here when error is not nil, otherwise the statement changes will be add to the next transaction. @jackysp

Comment (jackysp) [reply executor/write.go]: I think there is a cleanup in `StmtCommit`
https://github.com/pingcap/tidb/blob/7aefd5067ecddefb61d9def7a112e90d590bde0f/session/txn.go#L339
Will it happen?

**Comment by jackysp** [comment]
PTAL @tiancaiamao

**Comment by jackysp** [comment]
Maybe we need to implement this feature on release-2.1 directly. So DNM.

**Pull request closed without merging**